app/src/main/kotlin/trees/dataBases/RBT/Neo4jRepository.kt

```diff
@@ -0,0 +1,100 @@
+package trees.dataBases.RBT
+
+import org.neo4j.driver.*
+import org.neo4j.driver.exceptions.SessionExpiredException
+import org.neo4j.driver.exceptions.value.Uncoercible
+import trees.nodes.RBNode
+import java.io.Closeable
+import java.io.IOException
+
+class Neo4jRepository: Closeable {
```

Comment:
1. Пользуйтесь автоформатированием IDE:
```suggestion
class Neo4jRepository : Closeable {
```
2. Для работы с `Closable` ресурсами (например, сессиями) используйте `use` -- вам не придется руками их закрывать, это обезопасит вас от утечек.
3. У вас бросаются эксепшены -- убедитесь, что приложение будет их отлавливать и выдавать пользователю понятное сообщение так, чтобы само приложение при этом не падало.